pocs/utils/database.py

```diff
@@ -8,6 +10,108 @@
 from pocs.utils import current_time
 from pocs.utils import serializers as json_util
 from pocs.utils.config import load_config
+from pocs.utils.logger import get_root_logger
+
+
+class AbstractPanDB(metaclass=abc.ABCMeta):
+    def __init__(self, db_name=None, collection_names=list(), logger=None, **kwargs):
+        """
+        Init base class for db instances.
+
+        Args:
+            db_name: Name of the database, typically panoptes or panoptes_testing.
+            collection_names (list of str): Names of the valid collections.
+            logger: (Optional) logger to use for warnings.
+        """
+        self.db_name = db_name
+        self.collection_names = collection_names
+        self.logger = logger
+
+    def _warn(self, *args, **kwargs):
+        if self.logger:
+            self.logger.warning(*args, **kwargs)
+        else:
+            warn(*args)
+
+    def validate_collection(self, collection):
+        if collection not in self.collection_names:
+            msg = 'Collection type {!r} not available'.format(collection)
```

Comment:
Necessary? No. I prefer it because it puts quotes around it if it is a string; for example, an empty string would appear as "", which is a lot easier to read in the whole message, which then would be `Collection type "" not available`, as opposed to `Collection type  not available`.

Reply:
As per our private conversation, let's maybe look at wrapping this up into StrFormatLogRecord. Fine to leave in this PR for now.